examples/cloudsuite/kit/cloudsuite.py

```diff
@@ -0,0 +1,475 @@
+            print_output=False,
+        )
+
+    def parse_output_to_results(  # pylint: disable=arguments-differ
+        self,
+        command_output: str,
+        benchmark_duration_seconds: int,
+        record_data_dir: PathType,
+        **kwargs,
+    ) -> Dict[str, Any]:
+
+        thr = re.search(
+            r".*<driverSummary.*?"
+            r"<metric unit=(.*?)>(?P<throughput>.*?)<"
+            r".*?<startTime>(?P<time>.*?)<"
+            r".*?<totalOps unit=(.*?)>(?P<operations>.*?)<"
+            r".*?<rtXtps>(?P<rtXtps>.*?)<",
+            command_output,
+            flags=re.DOTALL,
+        )
+
+        pattern = r"<operation name=(.*?)>.*?" r"avg(.*?)/.*?" r"max(.*?)/.*?" r"sd(.*?)/"
+
+        m = re.findall(
+            pattern,
+            command_output,
+            flags=re.DOTALL,
+        )
+
+        add_to_csv = {
+            "run": thr.group("time"),
+            "throughput": thr.group("throughput"),
+            "operations": thr.group("operations"),
+            "rtXtps": thr.group("rtXtps"),
+        }
+
+        for op in m:
+            oper, o_avg, o_max, o_sd = op
+            oper = oper[1:-1]
+
+            if o_avg != "":
+                o_avg = o_avg[1:-1]
+                o_max = o_max[1:-1]
+                o_sd = o_sd[1:-1]
+            else:
+                o_avg = ""
+                o_max = ""
+                o_sd = ""
+
+            add_to_csv[f"{oper}_average"] = o_avg
+            add_to_csv[f"{oper}_max"] = o_max
+            add_to_csv[f"{oper}_sd"] = o_sd
+
+        self.get_details(record_data_dir)
+
+        self._write_to_record_data_dir(command_output, "full_output.txt", record_data_dir)
+
+        return add_to_csv
+
+
+def cloudsuite_campaign(
+    name: str = "cloussuite_campaign",
+    benchmark: Optional[CloudsuiteBench] = None,
+    src_dir: Optional[PathType] = None,
+    results_dir: Optional[PathType] = None,
+    command_wrappers: Iterable[CommandWrapper] = (),
+    command_attachments: Iterable[CommandAttachment] = (),
+    shared_libs: Iterable[SharedLib] = (),
+    pre_run_hooks: Iterable[PreRunHook] = (),
+    post_run_hooks: Iterable[PostRunHook] = (),
+    server_platform: Platform | None = None,
+    web_server_platform: Platform | None = None,
+    platform: Platform | None = None,
+    nb_runs: int = 1,
+    benchmark_duration_seconds: int = 5,
+    nb_threads: Iterable[int] = (1,),
+    generator_seeds: Iterable[int] = (0,),
+    debug: bool = False,
+    gdb: bool = False,
+    enable_data_dir: bool = False,
+    continuing: bool = False,
+    constants: Optional[Dict[str, Any]] = None,
+    pretty: Optional[Dict[str, str]] = None,
+) -> CampaignCartesianProduct:
+    """Return a cartesian product campaign configured for the Cloudsuite benchmark."""
+
+    variables = {
+        "nb_threads": nb_threads,
+        "generator_seed": generator_seeds,
+    }
+
+    if src_dir is None:
+        raise ValueError("A src_dir argument must be defined manually.")
```

Comment:
please specify of which repo. For now the user is left without information of what src it must provide.

Reply:
Changed this message

Reply:
it is still unclear IMHO. This is for the `faban` repo, isn't it?

Reply:
No, this is for the cloudsuite repository. I've just added a link to the github in the error message